### PR TITLE
[DOPS-1161] Add envoy-filter template to replace proxy-body-size annotation

### PR DESCRIPTION
**Pull request by greykurotsuki**

### CODE REVIEW

**Review by volkorny — APPROVED** [review]

**Review by volodymyr-mykhailyk — CHANGES_REQUESTED** [review]
From my understanding - this change will create such a filter for every project. Let's not do this and instead create filter only when someone specified body size limit. Also - please generate golden test for this use case.

File: stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.values.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    version: "0.0.0"
+
+destinations:
+  - component: webserver
+
+ingress:
+  - host: leads-broker-st1.matic.com
+    gateway: external-matic-com
+    destination: webserver
+    max_request_bytes: "67108864"
```
Comment (volodymyr-mykhailyk) [inline stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.values.yaml]: Why is this value a string? Is it required? What happens if I don't specify quotes

Comment (volodymyr-mykhailyk) [inline stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.values.yaml]: I would recommend adding one level of nesting here and making a key a bit more clear:
```
- host: leads-broker.matic.com
   settings:
     max_request_size_bytes: 1234
```

File: stable/application-mesh/tests/golden/golden_test.go [hunk 2]
```diff
@@ -69,6 +69,11 @@ func TestMeshGoldenTemplates(t *testing.T) {
 	        ValuesFiles: []string{"components/service-entry.values.yaml"},
 	        Templates: []string{"templates/service-entry.yaml"},
 	    },
+		{
+	        GoldenFileName: "components/envoy-filter-max-body-size.golden.yaml",
+	        ValuesFiles: []string{"components/envoy-filter-max-body-size.values.yaml"},
+			Templates: []string{"templates/envoy-filter.yaml", "templates/virtual-service.yaml"},
```
Comment (volodymyr-mykhailyk) [inline stable/application-mesh/tests/golden/golden_test.go]: Something is off with indent

File: stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.golden.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+---
```
Comment (volodymyr-mykhailyk) [inline stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.golden.yaml:1]: While this is a component test - it includes other templates. Something is off

**Review by volodymyr-mykhailyk — APPROVED** [review]
Looks good. Please take a look at my latest comments

File: stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.values.yaml (hunk 1 above)
Comment (greykurotsuki) [reply stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.values.yaml]: It is necessary to use quotes to specify the exact value. For instance, when you specify `max_request_bytes: 67108864`, Helm translates it to `max_request_bytes: 6.7108864e+07`.

File: stable/application-mesh/tests/golden/golden_test.go (hunk 2 above)
Comment (greykurotsuki) [reply stable/application-mesh/tests/golden/golden_test.go]: Thanks for noticing.

File: stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.values.yaml (hunk 1 above)
Comment (volodymyr-mykhailyk) [reply stable/application-mesh/tests/golden/components/envoy-filter-max-body-size.values.yaml]: In this case, you should transform it to int representation when rendering. I assure you, next time someone edits this setting - they will forget to add quotes

**Pull request merged**